scripts/cmake/vcpkg_configure_make.cmake

```
@@ -225,6 +227,14 @@ macro(z_convert_to_list input output)
     string(REGEX MATCHALL "(( +|^ *)[^ ]+)" ${output} "${${input}}")
 endmacro()
 
+macro(z_vcpkg_setup_detected_env env_name cmake_var)
+    if(VCPKG_DETECTED_${cmake_var})
+        if(NOT VCPKG_TARGET_IS_OSX)
+            set(ENV{${env_name}} "${VCPKG_DETECTED_${cmake_var}}")
+        endif()
+    endif()
+endmacro()

```

This doesn't need to be a macro:

```suggestion
function(z_vcpkg_setup_detected_env env_name cmake_var)
    if(DEFINED VCPKG_DETECTED_${cmake_var})
        if(NOT VCPKG_TARGET_IS_OSX)
            set("ENV{${env_name}}" "${VCPKG_DETECTED_${cmake_var}}")
        endif()
    endif()
endfunction()
```

---

Yes, I agree with you. I don't know why most vcpkg functions are declared as macros. I think it is because the very first function was declared as a macro, then the following contributors follow the convention. So let's follow the convention either. Although functions are better, macros work fine either.